File: woodroid/src/main/java/me/gilo/woodroid/repo/CustomerRepository.java
package me.gilo.woodroid.repo;

import me.gilo.woodroid.data.api.CustomerAPI;
import me.gilo.woodroid.models.Customer;
import retrofit2.Call;

import java.util.List;

public class CustomerRepository extends WooRepository{

    private final CustomerAPI apiService;

    public CustomerRepository(String baseUrl, String consumerKey, String consumerSecret) {
       super(baseUrl, consumerKey, consumerSecret);
       apiService = retrofit.create(CustomerAPI.class);
    }

    public Call<Customer> create(Customer customer) {
        return apiService.create(customer);
    }


    public Call<Customer> customer(int id) {
        return apiService.view(id);
    }

    public Call<List<Customer>> customers() {
        return apiService.list();
    }

    public Call<Customer> update(int id, Customer customer) {
        return apiService.update(id, customer);
    }

    public Call<Customer> delete(int id) {
        return apiService.delete(id);
    }

    public Call<Customer> delete(int id, boolean force) {
        return apiService.delete(id, force);
    }


}

File: woodroid/src/main/java/me/gilo/woodroid/repo/CustomerRepository.kt
package me.gilo.woodroid.repo

import me.gilo.woodroid.data.api.CustomerAPI
import me.gilo.woodroid.models.Customer
import me.gilo.woodroid.models.filters.CustomerFilter
import retrofit2.Call

class CustomerRepository(baseUrl: String, consumerKey: String, consumerSecret: String) :
    WooRepository(baseUrl, consumerKey, consumerSecret) {

    private val apiService: CustomerAPI

    init {
        apiService = retrofit.create(CustomerAPI::class.java)
    }

    fun create(customer: Customer): Call<Customer> {
        return apiService.create(customer)
    }


    fun customer(id: Int): Call<Customer> {
        return apiService.view(id)
    }

    fun customers(): Call<List<Customer>> {
        return apiService.list()
    }

    fun customers(customerFilter: CustomerFilter): Call<List<Customer>> {
        return apiService.filter(customerFilter.filters)
    }

    fun update(id: Int, customer: Customer): Call<Customer> {
        return apiService.update(id, customer)
    }

    fun delete(id: Int): Call<Customer> {
        return apiService.delete(id)
    }

    fun delete(id: Int, force: Boolean): Call<Customer> {
        return apiService.delete(id, force)
    }


}
